merkle/sparse_merkle_tree_test.go

```diff
@@ -231,17 +241,21 @@ func TestInclusionProofGetsIncorrectNode(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
 
-	const rev = 100
-	r, tx := getSparseMerkleTreeReaderWithMockTX(mockCtrl, rev)
-	tx.EXPECT().Commit().AnyTimes().Return(nil)
-	tx.EXPECT().GetMerkleNodes(int64(rev), gomock.Any()).Return([]storage.Node{getRandomNonRootNode(t, 34)}, nil)
-	const key = "SomeArbitraryKey"
-	_, err := r.InclusionProof(rev, []byte(key))
-	if err == nil {
-		t.Fatal("InclusionProof() should've returned an error due to incorrect node from storage layer")
-	}
-	if !strings.Contains(err.Error(), "1 remain(s) unused") {
-		t.Fatalf("Saw unexpected error: %v", err)
+	// This test requests inclusion proofs where storage returns a single node that should not be part
+	// of the proof for the supplied key. This should not succeed.
+	for _, testNode := range inclusionProofIncorrectTestNodes {
+		const rev = 100
+		r, tx := getSparseMerkleTreeReaderWithMockTX(mockCtrl, rev)
+		tx.EXPECT().Commit().AnyTimes().Return(nil)
+		tx.EXPECT().GetMerkleNodes(int64(rev), gomock.Any()).Return([]storage.Node{testNode}, nil)
+		const key = "SomeArbitraryKey"
+		proof, err := r.InclusionProof(rev, []byte(key))
+		if err == nil {
+			t.Fatalf("InclusionProof() should've returned an error due to incorrect node from storage layer: %v", proof)
```

Comment:
As we're making changes could we fix the pre-existing: 
 - no need to be fatal
 - prefer canonical form for error messages.